src/main/java/com/CarRentalAgency/services/DealerServiceImpl.java

```diff
@@ -42,13 +42,11 @@ public Dealer saveDealer(Dealer dealer) {
         // check if the dealer with the same email already exists
         Dealer existingDealer = dealerRepository
                 .findByEmail(dealer.getEmail()) // return the dealer with the specified email
-                .orElse(null); // if the dealer does not exist return null
+                .orElseThrow(() -> new AlreadyExistsException("THIS EMAIL:" + dealer.getEmail() + " ALREADY USED !!"));
 
         // if the email is not used by another dealer, save the dealer
-        if (existingDealer == null) {
             dealerRepository.save(dealer);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `saveDealer` method saves the dealer without checking if the dealer already exists. This could lead to data inconsistency if two dealers with the same email are saved. The check for the existence of the dealer should be done before saving.

```diff
- dealerRepository.save(dealer);
+ if (existingDealer == null) {
+     dealerRepository.save(dealer);
+ } else {
+     throw new AlreadyExistsException("THIS EMAIL:" + dealer.getEmail() + " ALREADY USED !!");
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->